unitTests/utils/UrlUtilsTest.ts

```diff
@@ -266,5 +266,57 @@ export function UrlUtilsTest() {
       });
       expect(url).toBe(`https://a.com?123=4&abc=d&efg=h`);
     });
+
+    describe('getLinkDestination', () => {
+      const originHost = 'https://cname.somewebsite.org';
+      const originPages = ['members', 'me'];
+      const originPage = originPages.join('/');
+      const origin = `${originHost}/${originPage}?friends=0&favorite-cake=chocolate#blog-title`;
+      it('should treat a path starting with a slash as absolute', () => {
+        expect(UrlUtils.getLinkDestination(origin, '/')).toEqual(`${originHost}/`);
+        expect(UrlUtils.getLinkDestination(origin, '/hello/world')).toEqual(`${originHost}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, '/hello/world/')).toEqual(`${originHost}/hello/world/`);
+      });
+
+      it('should treat a path not starting with a slash as relative', () => {
+        expect(UrlUtils.getLinkDestination(origin, '')).toEqual(`${originHost}/${originPage}`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/world')).toEqual(`${originHost}/${originPage}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/world/')).toEqual(`${originHost}/${originPage}/hello/world/`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/world/')).toEqual(`${originHost}/${originPage}/hello/world/`);
+      });
+
+      it('should correctly handle single periods', () => {
+        expect(UrlUtils.getLinkDestination(origin, '.')).toEqual(`${originHost}/${originPage}`);
+        expect(UrlUtils.getLinkDestination(origin, './')).toEqual(`${originHost}/${originPage}/`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/./world')).toEqual(`${originHost}/${originPage}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/world/.')).toEqual(`${originHost}/${originPage}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, 'hello/world/./')).toEqual(`${originHost}/${originPage}/hello/world/`);
+        expect(UrlUtils.getLinkDestination(origin, '/hello/./world')).toEqual(`${originHost}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, '/hello/world/.')).toEqual(`${originHost}/hello/world`);
+        expect(UrlUtils.getLinkDestination(origin, '/hello/world/./')).toEqual(`${originHost}/hello/world/`);
+      });
+
+      it('should correctly handle double periods', () => {
+        expect(UrlUtils.getLinkDestination(origin, '..')).toEqual(`${originHost}/${originPages[0]}`);
```

Comment:
very nitpicky here, but the index might be better as `originPages' last index - 1` or something along that line here.
This would make the test more resistant to changes; what if due to some tests in the future require `originPages` to have more elements in it (which is admittedly unlikely)?

Reply:
Good point, I'd change it for your suggestion but as it turns out, this code needs to be removed anyways.